test/util/malicious/test_app.go

```diff
@@ -0,0 +1,81 @@
+package malicious
+
+import (
+	"io"
+	"path/filepath"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/test/util"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/server"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/cosmos-sdk/snapshots"
+	snapshottypes "github.com/cosmos/cosmos-sdk/snapshots/types"
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cast"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+)
+
+func NewTestApp(cparams *tmproto.ConsensusParams, genAccounts ...string) *App {
+	app, _ := util.SetupTestAppWithGenesisValSet(cparams, genAccounts...)
+	badapp := &App{App: app}
+	badapp.SetPrepareProposalHandler(app.PrepareProposal)
+	return badapp
+}
+
+func NewAppServer(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts servertypes.AppOptions) servertypes.Application {
+	var cache sdk.MultiStorePersistentCache
```

Comment:
this function allows us to use the existing infrastructure (testnode, and even the cli if need be)